locust/contrib/geventhttpclient.py

```diff
@@ -61,23 +66,31 @@ class GeventHttpSession(object):
     def __init__(self, base_url):
         self.base_url = base_url
         self.cookiejar = CookieJar()
-        self.client = UserAgent(max_retries=1, cookiejar=self.cookiejar)
-        
+        self.client = LocustUserAgent(max_retries=1, cookiejar=self.cookiejar)
+    
+    def _build_url(self, path):
+        """ prepend url with hostname unless it's already an absolute URL """
+        if absolute_http_url_regexp.match(path):
+            return path
+        else:
+            return "%s%s" % (self.base_url, path)
```

Comment:
- do you need a regex here, or is `path.startswith('http')` sufficient?
- see `urllib.parse.urljoin()` (or `urlparse.urljoin()` in py2).  It handles edge cases (like extra slashes) better than just concatenating the pieces